### PR TITLE
[12.0][FIX] Improved code of stage create method.

### PR DESCRIPTION
**Pull request by rutul-serpentcs**
When I create stage from Field Service >> Dashboard >> Team >> Orders kanban view I got an error.

Teams-Odoo

### CODE REVIEW

**Review by brian10048 — CHANGES_REQUESTED** [review]
@rutul-serpentcs 
This fixes the original error you presented.
However there is still an issue creating the stage from Kanban view

```
The operation cannot be completed:
- Create/update: a mandatory field is not set.
- Delete: another model requires the record being deleted. If possible, archive it instead.

Model: Field Service Stage (fsm.stage), Field: Type (stage_type) 
```
stage_type field needs to be set

**Comment by rutul-serpentcs** [comment]
> @rutul-serpentcs
> This fixes the original error you presented.
> However there is still an issue creating the stage from Kanban view
> 
> ```
> The operation cannot be completed:
> - Create/update: a mandatory field is not set.
> - Delete: another model requires the record being deleted. If possible, archive it instead.
> 
> Model: Field Service Stage (fsm.stage), Field: Type (stage_type) 
> ```
> 
> stage_type field needs to be set

@brian10048 Done, Can you please review it again?

**Comment by brian10048** [comment]
@rutul-serpentcs - It allows to create the stage, but a new bug is introduced when we create a stage that has an undefined stage type.  When you go back to the orders kanban view, the stage recently created will not be displayed.

Can you revert the recent change and keep the stage_type required in python?  Then, instead can you use the context to set the correct `stage_type` based on the view

File: fieldservice/models/fsm_stage.py [hunk 1]
```diff
@@ -45,8 +45,7 @@ def _default_team_ids(self):
     stage_type = fields.Selection([('order', 'Order'),
                                    ('equipment', 'Equipment'),
                                    ('location', 'Location'),
-                                   ('worker', 'Worker')], 'Type',
-                                  required=True)
+                                   ('worker', 'Worker')], 'Type', require=True)
```
Comment (dreispt) [inline fieldservice/models/fsm_stage.py:48]: I'm afraid this is incorrect,  `required` is the correct field attribute.
Is the goal to remove the required attribute?

File: fieldservice/views/fsm_team.xml [hunk 2]
```diff
@@ -13,7 +13,8 @@
         <field name="name">Orders</field>
         <field name="res_model">fsm.order</field>
         <field name="view_mode">kanban,timeline,tree,form,calendar</field>
-        <field name="context">{'default_team_id': active_id}</field>
+        <field name="context">{'default_team_id': active_id,
+            'default_stage_type':'order'}</field>
```
Comment (dreispt) [inline fieldservice/views/fsm_team.xml:17]:
```suggestion
            'default_stage_type': 'order'}</field>
```

File: fieldservice/models/fsm_stage.py [hunk 3]
```diff
@@ -73,8 +72,8 @@ def get_color_information(self):
     def create(self, vals):
         stages = self.env['fsm.stage'].search([])
         for stage in stages:
-            if stage.stage_type == vals['stage_type'] and \
-               stage.sequence == vals['sequence']:
+            if stage.stage_type == vals.get('stage_type') and \
+               stage.sequence == vals.get('sequence'):
```
Comment (dreispt) [inline fieldservice/models/fsm_stage.py:76]: :+1:

**Review by max3903 — CHANGES_REQUESTED** [review]
Please attend the comments

**Comment by ivantodorovich** [comment]
I think we can close this one (or remove the "no stale" label), given it's old and the comments haven't been taken care of.

**Pull request closed without merging**